test/modules/attributes/AttributesController.test.ts

```diff
@@ -1222,6 +1222,127 @@ export class AttributesControllerTest extends IntegrationTest {
                     expect(attributes.length).to.equal(4)
                 })
             })
+
+            it("should return all versions of a possibly succeeded repository attribute", async function () {
+                const version0 = await consumptionController.attributes.createLocalAttribute({
+                    content: IdentityAttribute.from({
+                        value: {
+                            "@type": "Nationality",
+                            value: "DE"
+                        },
+                        owner: CoreAddress.from("address")
+                    })
+                })
+                const successorParams1: IAttributeSuccessorParams = {
+                    content: IdentityAttribute.from({
+                        value: {
+                            "@type": "Nationality",
+                            value: "US"
+                        },
+                        owner: CoreAddress.from("address")
+                    })
+                }
+                const successorParams2: IAttributeSuccessorParams = {
+                    content: IdentityAttribute.from({
+                        value: {
+                            "@type": "Nationality",
+                            value: "CZ"
+                        },
+                        owner: CoreAddress.from("address")
+                    })
+                }
+
+                const onlyVersion0 = await consumptionController.attributes.getVersionsOfAttribute(version0.id)
+                expect(onlyVersion0).to.eql([version0])
+
+                const { predecessor: updatedVersion0, successor: version1 } =
+                    await consumptionController.attributes.succeedRepositoryAttribute(version0.id, successorParams1)
+                const { predecessor: updatedVersion1, successor: version2 } =
+                    await consumptionController.attributes.succeedRepositoryAttribute(version1.id, successorParams2)
+
+                const allVersions0 = await consumptionController.attributes.getVersionsOfAttribute(version0.id)
+                expect(allVersions0).to.eql([version2, updatedVersion1, updatedVersion0])
```

Comment:
please use the long-form for equality everywhere
```suggestion
                expect(allVersions0).to.equal([version2, updatedVersion1, updatedVersion0])
```

Reply:
Using the long-form, the test fails, since the compared objects have the same content, but are two different objects.

Reply:
Changed it to deep.equal